### DISCUSSION TITLE
Issues with calcODF function when trying to produce a pole figure

### DISCUSSION THREAD

**Discussion by ginnyyb** [mtex-toolbox/mtex#1192] [Ask Anything] [answered]
After importing my EBSD data and successfully producing phase maps, IPF-Z maps + keys, and an inverse pole figure for all bytownite (plag) grains, I can not seem to get the function 'calcODF' to produce a pole figure. We keep getting these errors:

Error in orientation/calcODF (line 64)
  odf = calcFourierODF(ori,varargin{:},'kernel',psi);

Error in script19MP23 (line 256)
odf_bytownite = calcODF(ebsd(mergedGrains,'Bytownite').orientations,'halfwidth',10*degree);**

"mergedGrains" is a variable we created to merge grains previously separated by twin boundaries but this doesn't seem to be the issue - it seems to be something concerning the syntax of calcODF. 

After staring at this line for days and trying variations, I could really use some help.
Using MTEX 5.1.1 and MATLAB R2021a

Thank you,
G

**Comment by lfgmorales** [comment]
Hi,

You may want to update to MTEX 5.7.0, then instead of calcODF, try calcDensity(ebsd(mergedGrains(‚bytownite‘).orientations….), that should work, dont forget to have the same number of parentheses closing

Cheers

Luiz

Luiz F. G. Morales
ScopeM - ETH Zürich
Sent from iPhone

On 6 Sep 2021, at 21:58, ginnyyb ***@***.***> wrote:

﻿

After importing my EBSD data and successfully producing phase maps, IPF-Z maps + keys, and an inverse pole figure for all bytownite (plag) grains, I can not seem to get the function 'calcODF' to produce a pole figure. We keep getting these errors:

Error in orientation/calcODF (line 64)
odf = calcFourierODF(ori,varargin{:},'kernel',psi);

Error in script19MP23 (line 256)
odf_bytownite = calcODF(ebsd(mergedGrains,'Bytownite').orientations,'halfwidth',10*degree);**

"mergedGrains" is a variable we created to merge grains previously separated by twin boundaries but this doesn't seem to be the issue - it seems to be something concerning the syntax of calcODF.

After staring at this line for days and trying variations, I could really use some help.
Using MTEX 5.1.1 and MATLAB R2021a

Thank you,
G

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mtex-toolbox/mtex/discussions/1192>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/ALE5IN2IJWI5YFYH5FWX4E3UAUMNLANCNFSM5DQ75X7A>.

[ { ***@***.***": "http://schema.org", ***@***.***": "EmailMessage", "potentialAction": { ***@***.***": "ViewAction", "target": "https://github.com/mtex-toolbox/mtex/discussions/1192", "url": "https://github.com/mtex-toolbox/mtex/discussions/1192", "name": "View Discussion" }, "description": "View this Discussion on GitHub", "publisher": { ***@***.***": "Organization", "name": "GitHub", "url": "https://github.com" } } ]

**Reply by ginnyyb** [reply]
Thank you so much for the suggestion! I think you are certainly correct about needing to use MTEX 5.7.0 but unfortunately I am still getting the same error message...

**Comment by zmichels** [comment]
If you type the following into the command line (the same but without the call to either calcODF or calcDensity…) what is the result:

ebsd(mergedGrains,'Bytownite').orientations

**Reply by ginnyyb** [reply]
Thank you for the reply! It saved odf_qtz as a new variable and by habit I plotted it and got this as a result:
![Screen Shot 2021-09-06 at 4 35 59 PM](https://user-images.githubusercontent.com/90217800/132265925-6099f26d-8f9b-43ff-b610-a8acba000fb6.png)

**Reply by zmichels** [reply]
But what about:

ebsd(mergedGrains,'Bytownite').orientations

?

(Not 'quartz')

**Reply by ginnyyb** [reply]
Oops, should have removed that instead of just commenting it out. I had been trying to get an ODF for bytownite for a different sample in another script (but was also having the same issues there). I have only been using this script for my quartz grains as to not over-write any of my variables

**Reply by zmichels** [reply]
Are you able to compute an ODF from the quartz orientations then? 

I'm asking because I suspect something is odd about the data, rather than the calcODF/calcDensity functions... since everyone using MTEX has the same functions (and they work), but they don't have your data.

**Reply by ginnyyb** [reply]
I see my mistake here... I accidentally copy/pasted and uploaded the error message for the Bytownite function instead of the Quartz one, sorry about that confusion. I cannot get it to work for either phase for the 3 samples I have tried so far, so it could very well be an issue with the data.

I am importing .crc files and using the function EBSD.load to contain the data if this helps (sorry, I am very new to MTEX)

**Reply by zmichels** [reply]
If you feel comfortable, can you upload one of your .CRC files and your script you are using? My guess is that the raw data are fine, but that something went awry at the merging-of-grains stage of your script or somewhere else, and having all the pieces will help to diagnose.

Cheers,
Zach

**Reply by ginnyyb** [reply]
Yes, absolutely! Unfortunately, this forum will not allow me to upload these file types (even compressed into a ZIP file). I hope this is not over-stepping, is there any way I can email the files to you? I see you have your email listed on your profile and figured this would be the next best thing

**Comment by zmichels** [comment]
Sure, I don’t mind — I’ll try to look at it tonight. Alternatively, if you put them on a file-sharing site and post a link, then others can play along. Either way is good.

Cheers,
Zach
***@***.***

**Comment by ginnyyb** [comment, marked as answer]
Update: the problem with the script was not the syntax but MTEX had initially downloaded incorrectly. My computer had blocked a few applications imbedded into the software as 'malware' - to fix this issue I re-downloaded + re-installed MTEX 5.7.0 and manually blocked the pop-ups within my privacy settings (on a macbook).
